backend/django/create-model-and-api.mo.md

```diff
@@ -0,0 +1,219 @@
+    description = models.TextField(max_length=1000)   #TextField is a scrollable paragraph in the admin
+    created_at = models.DateTimeField(auto_now_add=True)    #auto_now_add generates automatically a timestamp when you create your create the news
+    updated_at = models.DateTimeField(auto_now=True)    #auto_now updates the timestamp everytime you change the news
+    city = models.ForeignKey('locations.city')    #All the cities of the City model of the locations app
+
+    class Meta:
+        verbose_name_plural = "News"    #The name of the class should be singular, this forces the plural to be News and not Newss
+
+    def __str__(self):
+        return self.title
+```
+
+
+- If you have users that are in groups, they need the permission to access to the Model. We won't explain how to create a group, we just want to show you how to add permissions to them. They will be added automaticcaly when you deploy.
+
+ouproject/users/apps/py
+```python
+from django.apps import AppConfig
+from django.contrib.admin import site
+from django.db.models.signals import post_migrate
+
+
+class UsersConfig(AppConfig):
+    name = 'intramuros.users'
+    verbose_name = "Users"
+
+    def ready(self):
+        post_migrate.connect(add_group_permissions, sender=self)
+
+
+def add_model_permissions(group, model, ContentType, Permission):
+    content_type = ContentType.objects.get_for_model(model)
+    permissions = Permission.objects.filter(content_type=content_type)
+    for permission in permissions:
+        group.permissions.add(permission)
+
+
+def add_group_permissions(sender, using, apps, **kwargs):
+    Group = apps.get_model("auth", "Group")
+    Permission = apps.get_model("auth", "Permission")
+    ContentType = apps.get_model("contenttypes", "ContentType")
+    News = apps.get_model("publications", "News")   #Add this line
+
+    if Group.objects.using(using).filter(name='Mayor Admin').exists():
+        group = Group.objects.get(name='Mayor Admin')
+    else:
+        group = Group.objects.using(using).create(name='Mayor Admin')
+    add_model_permissions(group, News, ContentType, Permission) #Add this second line
+```
+
+- Finally, make your migrations and migrate.
+
+
+### Add the News model to the admin *(~2 min)*
+
+- We now need to add the model to the admin back-office.
+
+ourproject/publications/admin.py
+```python
+from django.contrib import admin
+from .models import News
+
+
+@admin.register(News)
+class MyNewsAdmin(admin.ModelAdmin):
+    list_display = (
+        'title',
+        'city'
+    )
+```
+
+If you don't override `list_display` by writing `pass` instead of `list_display`, the default column title in the admin will be the __str__ returned in the model.
+list_display allows you to add several columns to the admin interface.
+
+
+### Serialize the News you get from the database *(~2 min)*
+
+- Serializers define the API representation.
```

Comment:
What does it do from the database to the viewset exactly?

Reply:
The serializer is the interface between the Django object (a model with an external key for instance) and a json object (shall I get the ID of the external key, or shall I use a nested object to describe the related city ?).

Then there is the View (or viewset, multiple views declared in the same object. The GET, and POST will be defined in the same viewset object). It defines the interface with the user: 
- Shall the user be authenticated?
- Does he have access to all the new?
- What happens if he posts a news: I should get the id of the user, then get his city, then create the news